src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -516,20 +516,8 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                     }
                 }
 
-                var arch = await MICommandFactory.GetTargetArchitecture();
-                if (arch == TargetArchitecture.Unknown)
-                {
-                    if (LaunchOptions.TargetArchitecture != TargetArchitecture.Unknown)
-                    {
-                        arch = LaunchOptions.TargetArchitecture;
-                    }
-                    else
-                    {
-                        WriteOutput(ResourceStrings.Warning_UsingDefaultArchitecture);
-                        arch = TargetArchitecture.X64;  // use as default
-                    }
-                }
-                SetTargetArch(arch);
+                // Must wait to get the target architecture until the first break event.
+				EnqueueInternalBreakAction(ObtainTargetArchitecture);
```

Comment:
>ObtainTargetArchitecture [](start = 31, length = 24)

This winds up setting the 'Is64BitArch' property which is used before the first time we enter break state (ex: AD7Module.cs(88)). I wonder if we need a different way of detecting this.